File: Examples/Efficiency_Evaluation_Examples/C_Programs/04-toylinarith1.c
//#Unsafe
// ****************************************************
//
//     Making Prophecies with Decision Predicates
//
//              Byron Cook * Eric Koskinen
//                     July 2010
//
// ****************************************************

//@ ltl invariant positive: AP(c > 5) ==> <>AP(resp > 5);

extern void __VERIFIER_error() __attribute__ ((__noreturn__));
extern void __VERIFIER_assume();
extern int __VERIFIER_nondet_int();

int c;
int servers ;
int resp;
int curr_serv;


c = 6; 
servers = 4;
resp = 0;
curr_serv = servers;


int main() {
	int tmp;
  while(curr_serv > 0) {
    if(1) {
      c--; curr_serv--;
      resp++;
    } else {
      curr_serv--;
    }
  }
  tmp=10000000;
  while (tmp>0) {tmp=tmp-1;}
}

File: Examples/Efficiency_Evaluation_Examples/MSVL_Programs/04-toylinarith1.m
//Unsafe

//AP(c > 5) ==> <>AP(resp > 5)
</
 define p: c>5;
 define q: resp>5;
 p->som(q)
/>
frame(c,servers,resp,curr_serv,tmp) and
(
int c and
int servers and
int resp and
int curr_serv and
int tmp and skip;

c:=6;   
servers := 4 and
resp := 0 and
curr_serv := servers;

  while(curr_serv > 0) {
	if(1) then
	{
	  c:=c-1;
      curr_serv:=curr_serv-1;
      resp:=resp+1
	}
    else
	{
	  curr_serv:=curr_serv-1
	}
  };
  tmp:=10000000;
  while(tmp>0)
  {
	tmp:=tmp-1
  }
)